### PR TITLE
add naivedatetime backward faker

### PR DESCRIPTION
**Pull request by pharosproduction**
Useful for database seed. For example post should be created later than user.

### CODE REVIEW

**Comment by tobyhinloopen** [comment]
I like your implementation better compared to the old implementation. However, there are now 2 implementations that do virtually the same.

I'm not the owner of the project so you can just completely ignore me, but I'm going to guess what kind of feedback you might get:

- [ ] Create a test for your new function(s). They don't have to test the randomness, just test if the function does anything and to show a valid use-case
- [ ] Consider creating a `forward` variant too
- [ ] Consider refactoring the code so the logic is shared between all combinations of `forward` and `backward`.
- [ ] Consider formatting the piped statements where the statements after the `=` sign is moved to the line below, indented with 2 spaces. See example below

```elixir
foo =
  bar()
  |> Baz.bam()
```

When you're done doing that, you should

- [ ] Add yourself to CONTRIBUTORS.md
- [ ] Add your change to CHANGELOG.md
- [ ] Add your addition to the USAGE.md

File: lib/faker/date.ex [hunk 1]
```diff
@@ -35,6 +35,22 @@ if Version.match?(System.version(), ">= 1.3.0") do
     end
 
     @doc """
+    Returns a random date in the past up to today in NaiveDateTime
```
Comment (tobyhinloopen) [inline lib/faker/date.ex]: Consider changing this line of doc to something like `Returns a random date between the given date in the past and today, today (not?) included`

**Comment by igas** [comment]
Thanks for contribution @pharosproduction!
Can you please take a look at @tobyhinloopen feedback, all points are valid.

**Comment by yordis** [comment]
@pharosproduction Are you still interested on contribute? I need this 😄 I can update the commit

File: lib/faker/date.ex [hunk 2]
```diff
@@ -35,6 +35,48 @@ if Version.match?(System.version(), ">= 1.3.0") do
...
+    """
+    @spec backward(NaiveDateTime.t) :: NaiveDateTime.t
+    def backward(date = %NaiveDateTime{}) do
```
Comment (anthonator) [inline lib/faker/date.ex]: What's the expected behavior when a date in the future is provided?

File: lib/faker/date.ex [hunk 3]
```diff
@@ -35,6 +35,48 @@ if Version.match?(System.version(), ">= 1.3.0") do
     end
 
     @doc """
+    Returns a random date in the past up to today in NaiveDateTime
+    """
+    @spec backward(NaiveDateTime.t) :: NaiveDateTime.t
```
Comment (anthonator) [inline lib/faker/date.ex]: Does it seem inconsistent to return `NaiveDateTime` for this function when all other functions in module return `Date`? Would it make more sense to have `Faker.DateTime` and `Faker.NaiveDateTime` modules?

**Pull request closed without merging**